test/replication-luatest/gh_5158_invalid_recovery_from_xlog_test.lua

```diff
@@ -0,0 +1,78 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+local fio = require('fio')
+local log = require('log')
+
+local g = t.group('gh_5158')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new({})
+    local box_cfg = {
+        checkpoint_count = 5,
+        log_level = 6,
+    }
+
+    cg.master = cg.cluster:build_server({alias = 'master', box_cfg = box_cfg})
+
+    box_cfg = {
+        replication = server.build_instance_uri('master'),
+    }
+
+    cg.replica = cg.cluster:build_server({alias = 'replica', box_cfg = box_cfg})
+    pcall(log.cfg, {level = 6})
+    cg.cluster:add_server(cg.master)
+    cg.cluster:add_server(cg.replica)
+    cg.cluster:start()
+end)
+
+g.after_each(function(cg)
+    cg.cluster:drop()
+end)
+
+g.test_invalid_recovery_from_xlog = function(cg)
+    cg.replica:stop()
+    t.helpers.retrying({}, function()
+        t.assert_str_matches(cg.master:exec(function()
+            return box.info.replication[2].downstream["status"]
+        end), 'stopped')
+    end)
+
+    cg.master:exec(function()
+        box.schema.space.create("test")
+        box.snapshot()
+        box.space.test:create_index("pk")
+        box.snapshot()
+    end)
+
+    cg.replica:start()
+
+    -- Vclocks are equal 5 on the master and on the replica
+    t.helpers.retrying({}, function()
+        t.assert_equals(cg.master:exec(function()
+            return box.info.vclock
+        end), {5})
+        t.assert_equals(cg.replica:exec(function()
+            return box.info.vclock
+        end), {5})
```

Comment:
Let's replace this with `cg.replica:wait_vclock_of(cg.master)`.
It's better to avoid waiting for specific values, like you do here (vclock is equal to 5). Later on, when something is changed in how tarantool writes WALs, vclocks will be equal to 6, or 4, or anything else, and the test will fail or will pass but become meaningless.

Reply:
I agree